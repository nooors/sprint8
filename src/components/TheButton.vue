<template>
  <div class="button" @click="sendInfo">
      <button>
          <slot></slot>
      </button>
  </div>
</template>

<script>
export default {
    name: 'TheButton',
    methods: {
        sendInfo(){
            return this.$emit('buttonClick');
        },
    }

}
</script>

<style lang="sass" scoped>
button
  background-color: black
  color: hsl(0, 0%, 52%)
  padding: .75rem
  border: 0
  border-radius: .5rem
  cursor: pointer
  transition: all 200ms
  &:hover
    box-shadow: 1px 1px 5px 2px #444242
    color: white
    text-shadow: 0 0 5px rgb(255, 255, 255)
</style>