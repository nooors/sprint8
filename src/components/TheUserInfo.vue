<template>
  <div>
    <div class="user__info">
      <div class="user__info__title">Welcome</div>
      <div class="user__info__name">{{ name }}</div>
      <div class="user__info__name">{{ surname }}</div>
      <div class="user__info__name">{{ displayName }}</div>
      <div class="user__info__name">{{ email }}</div>
      <div class="user__info__name__gap"></div>
      <button class="user__info__logout" @click="close">
        <img
          src="../assets/icons8-power-off-button-100.png"
          alt=""
          class="user__info__logout__img"
        />
      </button>
    </div>
  </div>
</template>

<script>
export default {
  name: "TheUserInfo",
  props: ["name", "surname", "displayName", "email"],
  methods: {
    close: function () {
      this.$emit("click");
    },
  },
};
</script>

<style lang="sass" scoped>
.user__info
    background-color: rgb(23, 23, 23)
    width: 12rem
    display: flex
    flex-direction: column
    border-radius: 0 0 .5rem .5rem

    .user__info__title
        background-color: rgb(222, 206, 28)
        color: rgb(23, 23, 23)
        text-align: center
        font-size: 1.5rem
        padding: .5rem 0
        text-transform: uppercase
        font-weight: bold
        border-radius: 0 0 .5rem .5rem

    .user__info__name
        color: rgb(222, 206, 28)
        padding: 1rem 0 0 .5rem
        border-bottom: 1px solid rgba(222, 206, 28, .2)
        border-radius: 0 0 .5rem .5rem
        background-color: rgb(23, 23, 23)

    .user__info__name__gap
        margin-bottom: .5rem

    .user__info__logout
        background-color: rgb(222, 206, 28)
        border: none
        border-top: 1px solid rgba(23, 23, 23, .2)
        border-radius: 0 0 .5rem .5rem
        cursor: pointer
        width: 100%

        .user__info__logout__img
            width: 2rem
            padding: .25rem 0
</style>
