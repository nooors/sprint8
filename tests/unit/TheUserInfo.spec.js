import { mount } from "@vue/test-utils";
import TheUserInfo from "@/components/TheUserInfo.vue";
import TheButton from "@/components/TheButton.vue";

describe("TheUserInfo component", () => {
  test("Renders the comoponent", () => {
    const wrapper = mount(TheUserInfo);
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("Render correct data", () => {
    const wrapper = mount(TheUserInfo, {
      propsData: {
        name: "Jhon",
        surname: "Doe",
        displayName: "DoePlayer",
        email: "doejhon@email.com",
      },
    });
    expect(wrapper.props().name).toBe("Jhon");
    expect(wrapper.props().surname).toBe("Doe");
    expect(wrapper.props().displayName).toBe("DoePlayer");
    expect(wrapper.props().email).toBe("doejhon@email.com");
  });
});
