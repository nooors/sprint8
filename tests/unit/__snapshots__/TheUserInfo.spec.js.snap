// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheUserInfo component Renders the comoponent 1`] = `
<div>
  <div class="user__info">
    <div class="user__info__title">Welcome</div>
    <div class="user__info__name"></div>
    <div class="user__info__name"></div>
    <div class="user__info__name"></div>
    <div class="user__info__name"></div>
    <div class="user__info__name__gap"></div> <button class="user__info__logout"><img src="../assets/icons8-power-off-button-100.png" alt="Logout" class="user__info__logout__img"></button>
  </div>
</div>
`;
